parquet-column/src/main/java/org/apache/parquet/column/page/PageReadStore.java

```diff
@@ -40,4 +42,25 @@
    */
   long getRowCount();
 
+  /**
+   * Returns the indexes of the rows to be read/built. All the rows which index is not returned shall be skipped.
+   *
+   * @return the incremental iterator of the row indexes
+   * @throws NotInPageFilteringModeException
+   *           if page filtering mode is not active so the related information is not available
+   * @see #isInPageFilteringMode()
+   */
+  default PrimitiveIterator.OfLong getRowIndexes() {
```

Comment:
Most of the code uses fastutil instead of Java 8 primitive iterators. It would be better to use the same one everywhere. Feel free to open an issue to move to Java 8 and eliminate fastutil, but I don't think we should mix them.

Reply:
I've used the Java 8 primitive iterators because this interface is public and did not want to expose fastutil classes. However, this API is internal, so we might use fastutil as well.
Unfortunately, Java 8 only introduced the primitive iterators but non of the other primitive implementations that fastutil offers (primitive lists, sets, maps etc.). So, I don't think we can drop fastutil.
Do you think it is better to use the fastutil primitive iterators here?